src/libutil/sysutil.cpp

```diff
@@ -532,37 +534,25 @@ Sysutil::put_in_background(int, char*[])
 Sysutil::put_in_background(int argc, char* argv[])
 #endif
 {
-    // You would think that this would be sufficient:
-    //   pid_t pid = fork ();
-    //   if (pid < 0)       // Some kind of error, we were unable to background
-    //      return false;
-    //   if (pid == 0)
-    //       return true;   // This is the child process, so continue with life
-    //   // Otherwise, this is the parent process, so terminate
-    //   exit (0);
-    // But it's not.  On OS X, it's not safe to fork() if your app is linked
-    // against certain libraries or frameworks.  So the only thing that I
-    // think is safe is to exec a new process.
-    // Another solution is this:
-    //    daemon (1, 1);
-    // But it suffers from the same problem on OS X, and seems to just be
-    // a wrapper for fork.
-
 #if defined(__linux__) || defined(__GLIBC__)
     // Simplest case:
     // daemon returns 0 if successful, thus return true if successful
     return daemon(1, 1) == 0;
 
 #elif defined(__APPLE__) && TARGET_OS_OSX
-    std::string newcmd = std::string(argv[0]) + " -F";
-    for (int i = 1; i < argc; ++i) {
-        newcmd += " \"";
-        newcmd += argv[i];
-        newcmd += "\"";
+    if (getppid() == 1) {
+        return true;
     }
```

Comment:
Could you please add a comment here explaining how this early return based on the parent process ID is the reason you don't end up recursively spawning new iv's?